backend/controllers/workoutTemplatesController.js

```diff
@@ -158,9 +159,81 @@ const getFeed = async (req, res) => {
   }
 };
 
+const vote = async (req, res) => {
+  const { templateId } = req.params;
+  const { voteType } = req.body;
+  const userId = req.userId;
+
+  try {
+    // Check if the workout template exists
+    const workoutTemplate = await prisma.workoutTemplate.findUnique({
+      where: { id: parseInt(templateId) },
+    });
+
+    if (!workoutTemplate) {
+      return res.status(404).json({ error: "Workout template not found" });
+    }
+
+    // Check if the user has already voted on this template
+    const existingVote = await prisma.vote.findUnique({
+      where: {
+        userId_workoutTemplateId: {
+          userId,
+          workoutTemplateId: parseInt(templateId),
+        },
+      },
+    });
+
+    let updatedVote;
+    if (existingVote) {
+      updatedVote = await prisma.vote.update({
```

Comment:
should we check if the existing vote type is equal to the updated vote type? this might help save some repetitive updates to your database

Reply:
I think you're right, I will add an additional condition for
``` if ( existingVote && existingVote.voteType !== voteType )```